knack/cli.py

```diff
@@ -219,6 +219,7 @@ def invoke(self, args, initial_invocation_data=None, out_file=None):
                 if cmd_result and cmd_result.result is not None:
                     formatter = self.output.get_formatter(output_type)
                     self.output.out(cmd_result, formatter=formatter, out_file=out_file)
+                self.raise_event(EVENT_CLI_SUCCESSFUL_EXECUTE, result=cmd_result.result)
```

Comment:
Should we raise the event only when exit_code is 0?

Reply:
Yes. On the other hand, `EVENT_CLI_POST_EXECUTE` is raised unconditionally. 

https://github.com/microsoft/knack/blob/61057a3a92756f01701c9428959a2bc1185559bf/knack/cli.py#L233-L234

Reply:
Then should you check the `exit_code` from https://github.com/microsoft/knack/blob/61057a3a92756f01701c9428959a2bc1185559bf/knack/cli.py#L217

Reply:
Not really necessary. 

https://github.com/microsoft/knack/blob/61057a3a92756f01701c9428959a2bc1185559bf/knack/cli.py#L215

will raise an exception first if `exit_code` is not `0`. 

Also,

https://github.com/microsoft/knack/blob/61057a3a92756f01701c9428959a2bc1185559bf/knack/cli.py#L221

already assumes the command is successfully executed and `exit_code` is `0`.

Reply:
got it.